engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -313,4 +336,166 @@ private void processAggregateEmailsByAggregationKey(AggregationCommand aggregati
         }
     }
+
+    private void processBundleAggregation(List<AggregationCommand> aggregationCommands, Event aggregatorEvent) {
+        final String bundleName = aggregationCommands.get(0).getAggregationKey().getBundle();
+        // Patch event display name for event log rendering
+        Optional<Bundle> bundle = bundleRepository.getBundle(bundleName);
+        if (bundle.isPresent()) {
+            String eventTypeDisplayName = String.format("%s - %s",
+                aggregatorEvent.getEventTypeDisplayName(),
+                bundle.get().getDisplayName());
+            eventRepository.updateEventDisplayName(aggregatorEvent.getId(), eventTypeDisplayName);
+        }
+
+        Endpoint endpoint = endpointRepository.getOrCreateDefaultSystemSubscription(null, aggregatorEvent.getOrgId(), EndpointType.EMAIL_SUBSCRIPTION);
+
+        //Store every aggregated application data for each user
+        Map<User, List<ApplicationAggregatedData>> userData = new HashMap<>();
+
+        for (AggregationCommand applicationAggregationCommand : aggregationCommands) {
+            try {
+                Map<User, Map<String, Object>> applicationAggregatedContextByUser = emailAggregator.getAggregated(applicationAggregationCommand.getAggregationKey(),
+                    applicationAggregationCommand.getSubscriptionType(),
+                    applicationAggregationCommand.getStart(),
+                    applicationAggregationCommand.getEnd());
+
+                applicationAggregatedContextByUser.entrySet().stream().forEach(userAggregation -> {
+                    userData.computeIfAbsent(userAggregation.getKey(), unused -> new ArrayList<>())
+                        .add(new ApplicationAggregatedData(userAggregation.getValue(), applicationAggregationCommand.getAggregationKey().getApplication()));
+                });
+            } catch (Exception ex) {
+                Log.error("Error processing " + applicationAggregationCommand.getAggregationKey(), ex);
+            }
+        }
+
+        // Group users with same aggregated data
+        Map<List<ApplicationAggregatedData>, Set<User>> usersWithSameAggregatedData = userData.keySet().stream()
+            .collect(Collectors.groupingBy(userData::get, Collectors.toSet()));
+
+        String emailTitle = "Daily digest - " + bundle.get().getDisplayName();
+
+        // for each set of users, generate email subject + body and send it to email connector
+        usersWithSameAggregatedData.entrySet().stream().forEach(listApplicationWithUserCollection -> {
+            Map<String, DailyDigestSection> dataMap  = new HashMap<>();
+            for (ApplicationAggregatedData applicationAggregatedData : listApplicationWithUserCollection.getKey()) {
+                try {
+                    generateAggregatedEmailBody(bundleName, applicationAggregatedData.appName, applicationAggregatedData.aggregatedData, dataMap);
```

Comment:
I'm not a big fan of collections passed as arguments to a method and modified while the method is executed.

`generateAggregatedEmailBody()` and `addItem()` could return a `DailyDigestSection` instance, which would then be added to `dataMap` here.